### PR TITLE
chore: rename sanitize functions for better searchability

### PR DESCRIPTION
**Pull request by rarkins**
## Changes:

Renames add/clear sanitization functions for better understandability and searchability.

## Context:

Part of #13807

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/util/host-rules.ts [hunk 1]
```diff
@@ -174,5 +174,5 @@ export function getAll(): HostRule[] {
 export function clear(): void {
   logger.debug('Clearing hostRules');
   hostRules = [];
-  sanitize.clear();
+  sanitize.clearSanitizedSecretsList();
```
Comment (viceice) [inline lib/util/host-rules.ts:177]: Can this be the cause for unsanitized secrets like platform password?

**Review by viceice — COMMENTED** [review]
Otherwise LGTM

File: lib/util/host-rules.ts (hunk 1 above)
Comment (rarkins) [reply lib/util/host-rules.ts:177]: Yes, it's part of the problem

File: lib/util/sanitize.ts [hunk 2]
```diff
@@ -26,11 +26,14 @@ export function sanitize(input: string): string {
   return output;
 }
 
-export function add(secret: string): void {
+export function addSecretForSanitizing(secret: string): void {
+  if (secret === '**redacted**') {
+    return;
+  }
```
Comment (rarkins) [inline lib/util/sanitize.ts:32]:
```suggestion
```

**Review by viceice — APPROVED** [review]

**Pull request merged**